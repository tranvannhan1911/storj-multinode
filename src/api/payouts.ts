// Copyright (C) 2021 Storj Labs, Inc.
// See LICENSE for copying information.

import { APIClient } from '@/api/index';
import { Expectation, HeldAmountSummary, NodePayoutsSummary, PayoutsSummary, Paystub } from '@/payouts';

/**
 * client for nodes controller of MND api.
 */
export class PayoutsClient extends APIClient {
    private readonly ROOT_PATH: string = '/api/v0/payouts';

    /**
     * Handles fetch of payouts summary information.
     *
     * @param satelliteId - satellite id.
     * @param period - selected period.
     *
     * @throws {@link BadRequestError}
     * This exception is thrown if the input is not a valid.
     *
     * @throws {@link UnauthorizedError}
     * Thrown if the auth cookie is missing or invalid.
     *
     * @throws {@link InternalError}
     * Thrown if something goes wrong on server side.
     */
    public async summary(satelliteId: string | null, period: string | null): Promise<PayoutsSummary> {
        let path = `${this.ROOT_PATH}`;

        if (satelliteId) {
            path += `/satellites/${satelliteId}`;
        }

        path += '/summaries';

        if (period) {
            path += `/${period}`;
        }

        const response = await this.http.get(path);

        if (!response.ok) {
            await this.handleError(response);
        }

        const result = await response.json();

        return new PayoutsSummary(
            result.totalEarned,
            result.totalHeld,
            result.totalPaid,
            result.nodeSummary.map(item => new NodePayoutsSummary(
                item.nodeId,
                item.nodeName,
                item.held,
                item.paid,
            )),
        );
    }

    /**
     * Handles fetch of payouts expectation such as estimated current month payout and undistributed payout.
     *
     * @param nodeId - node id.
     *
     * @throws {@link BadRequestError}
     * This exception is thrown if the input is not a valid.
     *
     * @throws {@link UnauthorizedError}
     * Thrown if the auth cookie is missing or invalid.
     *
     * @throws {@link InternalError}
     * Thrown if something goes wrong on server side.
     */
    public async expectations(nodeId: string | null): Promise<Expectation> {
        let path = `${this.ROOT_PATH}/expectations`;

        if (nodeId) {
            path += `/${nodeId}`;
        }

        const response = await this.http.get(path);

        if (!response.ok) {
            await this.handleError(response);
        }

        const result = await response.json();

        return new Expectation(
            result.currentMonthEstimation,
            result.undistributed,
        );
    }

    /**
     * Handles fetch of payouts paystub information for node.
     *
     * @param satelliteId - satellite id.
     * @param period - selected period.
     * @param nodeId
     *
     * @throws {@link BadRequestError}
     * This exception is thrown if the input is not a valid.
     *
     * @throws {@link UnauthorizedError}
     * Thrown if the auth cookie is missing or invalid.
     *
     * @throws {@link InternalError}
     * Thrown if something goes wrong on server side.
     */
    public async paystub(satelliteId: string | null, period: string | null, nodeId: string): Promise<Paystub> {
        let path = `${this.ROOT_PATH}`;

        if (satelliteId) {
            path += `/satellites/${satelliteId}`;
        }

        path += '/paystubs';

        if (period) {
            path += `/${period}`;
        }

        path += `/${nodeId}`;

        const response = await this.http.get(path);

        if (!response.ok) {
            await this.handleError(response);
        }

        const result = await response.json();

        return new Paystub(
            result.usageAtRest,
            result.usageGet,
            result.usageGetRepair,
            result.usageGetAudit,
            result.compAtRest,
            result.compGet,
            result.compGetRepair,
            result.compGetAudit,
            result.held,
            result.paid,
            result.distributed,
        );
    }

    /**
     * Handles fetch of payouts paystub information for node.
     *
     * @param nodeId
     *
     * @throws {@link BadRequestError}
     * This exception is thrown if the input is not a valid.
     *
     * @throws {@link UnauthorizedError}
     * Thrown if the auth cookie is missing or invalid.
     *
     * @throws {@link InternalError}
     * Thrown if something goes wrong on server side.
     */
    public async heldHistory(nodeId: string): Promise<HeldAmountSummary[]> {
        const path = `${this.ROOT_PATH}/held-amounts/${nodeId}`;

        const response = await this.http.get(path);

        if (!response.ok) {
            await this.handleError(response);
        }

        const result = await response.json();

        return result.map(heldHistoryItem => new HeldAmountSummary(
            heldHistoryItem.satelliteAddress,
            heldHistoryItem.firstQuarter,
            heldHistoryItem.secondQuarter,
            heldHistoryItem.thirdQuarter,
            heldHistoryItem.periodCount,
        ));
    }
}
