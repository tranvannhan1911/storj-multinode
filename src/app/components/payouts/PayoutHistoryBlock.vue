// Copyright (C) 2021 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <info-block>
        <div slot="body" class="payouts-history-block">
            <h3 class="payouts-history-block__title">Payout History</h3>
            <v-button class="payouts-history-block__button" label="Download" :is-white="true" width="100%" />
        </div>
    </info-block>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

import InfoBlock from '@/app/components/common/InfoBlock.vue';
import VButton from '@/app/components/common/VButton.vue';

// @vue/component
@Component({
    components: {
        InfoBlock,
        VButton,
    },
})
export default class PayoutHistoryBlock extends Vue {}
</script>

<style scoped lang="scss">
    .payouts-history-block {
        box-sizing: border-box;
        width: 100%;
        display: flex;
        flex-direction: column;
        align-items: flex-start;

        &__title {
            font-family: 'font_bold', sans-serif;
            font-size: 24px;
            color: var(--c-title);
            margin-bottom: 30px;
        }

        &__button {
            background-color: var(--c-button-gray);
            border: none !important;
        }
    }
</style>
