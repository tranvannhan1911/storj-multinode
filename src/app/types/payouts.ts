// Copyright (C) 2021 Storj Labs, Inc.
// See LICENSE for copying information.

import { monthNames } from '@/app/types/date';

/**
 * Describes month button entity for calendar.
 */
export class MonthButton {
    public constructor(
        public year: number = 0,
        public index: number = 0,
        public active: boolean = false,
        public selected: boolean = false,
    ) {}

    /**
     * Returns month label depends on index.
     */
    public get name(): string {
        return monthNames[this.index] ? monthNames[this.index].slice(0, 3) : '';
    }
}

export interface StoredMonthsByYear {
    [key: number]: MonthButton[];
}
