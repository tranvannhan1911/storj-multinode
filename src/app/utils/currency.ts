// Copyright (C) 2021 Storj Labs, Inc.
// See LICENSE for copying information.

/**
 * Size class contains currency related functionality such as convertation.
 */
export class Currency {
    /**
     * dollarsFromCents converts cents to dollars with prefix.
     * @param cents count
     */
    public static dollarsFromCents(cents: number): string {
        return `$${(cents / 100).toFixed(2)}`;
    }
}
