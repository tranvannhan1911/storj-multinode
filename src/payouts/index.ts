// Copyright (C) 2021 Storj Labs, Inc.
// See LICENSE for copying information.

/**
 * Divider to convert payout amounts to cents.
 */
const PRICE_DIVIDER = 10000;

/**
 * PayoutsSummary is a representation of summary of payout information for node.
 */
export class NodePayoutsSummary {
    public constructor(
        public nodeId: string = '',
        public nodeName: string = '',
        public held: number = 0,
        public paid: number = 0,
    ) {}

    /**
     * Returns node name for displaying.
     * If no provided returns id.
     */
    public get title(): string {
        return this.nodeName || this.nodeId;
    }
}

/**
 * PayoutsSummary is a representation of summary of payout information for all selected nodes and list of payout summaries by nodes.
 */
export class PayoutsSummary {
    public constructor(
        public totalEarned: number = 0,
        public totalHeld: number = 0,
        public totalPaid: number = 0,
        public nodeSummary: NodePayoutsSummary[] = [],
    ) {
        this.totalPaid = this.convertToCents(this.totalPaid);
        this.totalEarned = this.convertToCents(this.totalEarned);
        this.totalHeld = this.convertToCents(this.totalHeld);

        this.nodeSummary.forEach((summary: NodePayoutsSummary) => {
            summary.paid = this.convertToCents(summary.paid);
            summary.held = this.convertToCents(summary.held);
        });
    }

    private convertToCents(value: number): number {
        return value / PRICE_DIVIDER;
    }
}

/**
 * Represents payout period month and year.
 */
export class PayoutPeriod {
    public constructor(
        public year: number = new Date().getUTCFullYear(),
        public month: number = new Date().getUTCMonth(),
    ) {}

    public get period(): string {
        return this.month < 9 ? `${this.year}-0${this.month + 1}` : `${this.year}-${this.month + 1}`;
    }

    /**
     * Parses PayoutPeriod from string.
     * @param period string
     */
    public static fromString(period: string): PayoutPeriod {
        const periodArray = period.split('-');

        return new PayoutPeriod(parseInt(periodArray[0]), parseInt(periodArray[1]) - 1);
    }
}

/**
 * Contains all payout information of particular node.
 */
export class NodePayouts {
    public constructor(
        public totalEarned: number = 0,
        public totalHeld: number = 0,
        public totalPaid: number = 0,
        public heldHistory: HeldAmountSummary[] = [],
        public paystubForPeriod: Paystub = new Paystub(),
        public expectations: Expectation = new Expectation(),
    ) {}
}

/**
 * Holds held history information for all satellites.
 */
export class HeldAmountSummary {
    public constructor(
        public satelliteAddress: string = '',
        public firstQuarter: number = 0,
        public secondQuarter: number = 0,
        public thirdQuarter: number = 0,
        public periodCount: number = 0,
    ) {
        this.firstQuarter = this.convertToCents(this.firstQuarter);
        this.secondQuarter = this.convertToCents(this.secondQuarter);
        this.thirdQuarter = this.convertToCents(this.thirdQuarter);
    }

    /**
     * Returns node age depends on period count.
     */
    public get monthsCount(): string {
        return `${this.periodCount + 1} month${this.periodCount > 0 ? 's' : ''}`;
    }

    private convertToCents(value: number): number {
        return value / PRICE_DIVIDER;
    }
}

/**
 * Payouts month-term representation.
 */
export class Paystub {
    public gross = 0;

    public constructor(
        public usageAtRest: number = 0,
        public usageGet: number = 0,
        public usageGetRepair: number = 0,
        public usageGetAudit: number = 0,
        public compAtRest: number = 0,
        public compGet: number = 0,
        public compGetRepair: number = 0,
        public compGetAudit: number = 0,
        public held: number = 0,
        public paid: number = 0,
        public distributed: number = 0,
    ) {
        this.compAtRest = this.convertToCents(this.compAtRest);
        this.compGet = this.convertToCents(this.compGet);
        this.compGetRepair = this.convertToCents(this.compGetRepair);
        this.compGetAudit = this.convertToCents(this.compGetAudit);
        this.held = this.convertToCents(this.held);
        this.paid = this.convertToCents(this.paid);
        this.gross = this.convertToCents(this.paid + this.held);
        this.distributed = this.convertToCents(this.distributed);
    }

    public get repairAndAuditUsage(): number {
        return this.usageGetRepair + this.usageGetAudit;
    }

    public get repairAndAuditComp(): number {
        return this.compGetRepair + this.compGetAudit;
    }

    private convertToCents(value: number): number {
        return value / PRICE_DIVIDER;
    }
}

/**
 * Expectations is a representation of current month estimated payout and undistributed payouts.
 */
export class Expectation {
    public constructor(
        public currentMonthEstimation: number = 0,
        public undistributed: number = 0,
    ) {
        this.undistributed = this.convertToCents(this.undistributed);
    }

    private convertToCents(value: number): number {
        return value / PRICE_DIVIDER;
    }
}
