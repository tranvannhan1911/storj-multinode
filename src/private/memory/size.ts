// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

/**
 * Base10 sizes.
 */
export enum SizeBreakpoints {
    KB = 1e3,
    MB = 1e6,
    GB = 1e9,
    TB = 1e12,
    PB = 1e15,
    EB = 1e18,
}

/**
 * Size class contains size related functionality such as convertation.
 */
export class Size {
    /**
     * Base10String converts size to a string using base-10 prefixes.
     * @param size in bytes
     */
    public static toBase10String(size: number): string {
        const decimals = 2;

        const _size = Math.abs(size);

        switch (true) {
        case _size >= SizeBreakpoints.EB * 2 / 3:
            return `${parseFloat((size / SizeBreakpoints.EB).toFixed(decimals))}EB`;
        case _size >= SizeBreakpoints.PB * 2 / 3:
            return `${parseFloat((size / SizeBreakpoints.PB).toFixed(decimals))}PB`;
        case _size >= SizeBreakpoints.TB * 2 / 3:
            return `${parseFloat((size / SizeBreakpoints.TB).toFixed(decimals))}TB`;
        case _size >= SizeBreakpoints.GB * 2 / 3:
            return `${parseFloat((size / SizeBreakpoints.GB).toFixed(decimals))}GB`;
        case _size >= SizeBreakpoints.MB * 2 / 3:
            return `${parseFloat((size / SizeBreakpoints.MB).toFixed(decimals))}MB`;
        case _size >= SizeBreakpoints.KB * 2 / 3:
            return `${parseFloat((size / SizeBreakpoints.KB).toFixed(decimals))}KB`;
        default:
            return `${size ? size.toFixed(decimals) : size}B`;
        }
    }
}
