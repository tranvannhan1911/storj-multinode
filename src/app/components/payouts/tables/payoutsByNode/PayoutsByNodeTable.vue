// Copyright (C) 2021 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <base-table>
        <thead slot="head">
            <tr>
                <th class="align-left">NAME</th>
                <th>DISK</th>
                <th>BANDWIDTH</th>
                <th>PAYOUT</th>
            </tr>
        </thead>
        <tbody slot="body">
            <tr class="table-item">
                <th class="align-left">Download</th>
                <th>--</th>
                <th>{{ paystub.usageGet | bytesToBase10String }}</th>
                <th>{{ paystub.compGet | centsToDollars }}</th>
            </tr>
            <tr class="table-item">
                <th class="align-left">Download Repair</th>
                <th>--</th>
                <th>{{ paystub.repairAndAuditUsage | bytesToBase10String }}</th>
                <th>{{ paystub.repairAndAuditComp | centsToDollars }}</th>
            </tr>
            <tr class="table-item">
                <th class="align-left">Disk Space</th>
                <th>{{ paystub.usageAtRest | bytesToBase10String }}m</th>
                <th>--</th>
                <th>{{ paystub.compAtRest | centsToDollars }}</th>
            </tr>
            <tr class="table-item">
                <th class="align-left">Gross Total</th>
                <th /><th />
                <th>{{ paystub.gross | centsToDollars }}</th>
            </tr>
            <tr class="table-item">
                <th class="align-left">Held amount</th>
                <th /><th />
                <th>{{ paystub.held | centsToDollars }}</th>
            </tr>
            <tr class="table-item">
                <th class="align-left">NET TOTAL</th>
                <th /><th />
                <th>{{ paystub.paid | centsToDollars }}</th>
            </tr>
            <tr class="table-item">
                <th class="align-left">Distributed</th>
                <th /><th />
                <th>{{ paystub.distributed | centsToDollars }}</th>
            </tr>
        </tbody>
    </base-table>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

import { Paystub } from '@/payouts';

import BaseTable from '@/app/components/common/BaseTable.vue';

// @vue/component
@Component({
    components: {
        BaseTable,
    },
})
export default class PayoutsByNodeTable extends Vue {
    @Prop({ default: () => new Paystub() })
    public paystub: Paystub;
}
</script>
