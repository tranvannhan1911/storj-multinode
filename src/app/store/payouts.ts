// Copyright (C) 2021 Storj Labs, Inc.
// See LICENSE for copying information.

import { ActionContext, ActionTree, GetterTree, Module, MutationTree } from 'vuex';

import { RootState } from '@/app/store/index';
import { monthNames } from '@/app/types/date';
import { Expectation, HeldAmountSummary, NodePayouts, PayoutsSummary, Paystub } from '@/payouts';
import { Payouts } from '@/payouts/service';

/**
 * PayoutsState is a representation of payouts module state.
 */
export class PayoutsState {
    public summary: PayoutsSummary = new PayoutsSummary();
    public selectedPayoutPeriod: string | null = null;
    public selectedNodePayouts: NodePayouts = new NodePayouts();
    public selectedNodeExpectations: Expectation = new Expectation();
    public totalExpectations: Expectation = new Expectation();
}

/**
 * PayoutsModule is a part of a global store that encapsulates all payouts related logic.
 */
export class PayoutsModule implements Module<PayoutsState, RootState> {
    public readonly namespaced: boolean;
    public readonly state: PayoutsState;
    public readonly getters?: GetterTree<PayoutsState, RootState>;
    public readonly actions: ActionTree<PayoutsState, RootState>;
    public readonly mutations: MutationTree<PayoutsState>;

    private readonly payouts: Payouts;

    public constructor(payouts: Payouts) {
        this.payouts = payouts;

        this.namespaced = true;
        this.state = new PayoutsState();

        this.mutations = {
            setSummary: this.setSummary,
            setPayoutPeriod: this.setPayoutPeriod,
            setTotalExpectation: this.setTotalExpectation,
            setNodeTotals: this.setNodeTotals,
            setNodePaystub: this.setNodePaystub,
            setNodeHeldHistory: this.setNodeHeldHistory,
            setCurrentNodeExpectations: this.setCurrentNodeExpectations,
        };

        this.actions = {
            summary: this.summary.bind(this),
            expectations: this.expectations.bind(this),
            paystub: this.paystub.bind(this),
            nodeTotals: this.nodeTotals.bind(this),
            heldHistory: this.heldHistory.bind(this),
        };

        this.getters = {
            periodString: this.periodString,
        };
    }

    // Mutations
    /**
     * setSummary mutation will set payouts summary state.
     * @param state - state of the module.
     * @param summary - payouts summary information depends on selected time and satellite.
     */
    public setSummary(state: PayoutsState, summary: PayoutsSummary): void {
        state.summary = summary;
    }

    /**
     * setPayoutPeriod mutation will save selected period to store.
     * @param state
     * @param period representation of month and year
     */
    public setPayoutPeriod(state: PayoutsState, period: string | null): void {
        state.selectedPayoutPeriod = period;
    }

    /**
     * setTotalExpectation mutation will set total payouts expectation for all nodes.
     * @param state - state of the module.
     * @param expectations - payouts summary information depends on selected time and satellite.
     */
    public setTotalExpectation(state: PayoutsState, expectations: Expectation): void {
        state.totalExpectations = expectations;
    }

    /**
     * setNodeTotals mutation will total payout information for selected node to store.
     * @param state
     * @param paystub for all time
     */
    public setNodeTotals(state: PayoutsState, paystub: Paystub): void {
        state.selectedNodePayouts = { ...state.selectedNodePayouts, totalPaid: paystub.distributed, totalEarned: paystub.paid, totalHeld: paystub.held };
    }

    /**
     * setNodePaystub mutation will save paystub for selected node, satellite and period to store.
     * @param state
     * @param paystub
     */
    public setNodePaystub(state: PayoutsState, paystub: Paystub): void {
        state.selectedNodePayouts = { ...state.selectedNodePayouts, paystubForPeriod: paystub };
    }

    /**
     * setNodeHeldHistory mutation will save held history for selected node to store.
     * @param state
     * @param heldHistory
     */
    public setNodeHeldHistory(state: PayoutsState, heldHistory: HeldAmountSummary[]): void {
        state.selectedNodePayouts = { ...state.selectedNodePayouts, heldHistory };
    }

    /**
     * setCurrentNodeExpectations mutation will save paystub for selected node, satellite and period to store.
     * @param state
     * @param expectations
     */
    public setCurrentNodeExpectations(state: PayoutsState, expectations: Expectation): void {
        state.selectedNodePayouts = { ...state.selectedNodePayouts, expectations };
    }

    // Actions
    /**
     * summary action loads payouts summary information.
     * @param ctx - context of the Vuex action.
     */
    public async summary(ctx: ActionContext<PayoutsState, RootState>): Promise<void> {
        const selectedSatelliteId = ctx.rootState.nodes.selectedSatellite ? ctx.rootState.nodes.selectedSatellite.id : null;
        const summary = await this.payouts.summary(selectedSatelliteId, ctx.state.selectedPayoutPeriod);

        ctx.commit('setSummary', summary);
    }

    /**
     * expectations action loads payouts total or by node payout expectation information.
     * @param ctx - context of the Vuex action.
     * @param nodeId - node id.
     */
    public async expectations(ctx: ActionContext<PayoutsState, RootState>, nodeId: string | null): Promise<void> {
        const expectations = await this.payouts.expectations(nodeId);

        ctx.commit(`${nodeId ? 'setCurrentNodeExpectations' : 'setTotalExpectation'}`, expectations);
    }

    /**
     * paystub action loads payouts information for table for selected node.
     * @param ctx - context of the Vuex action.
     * @param nodeId
     */
    public async paystub(ctx: ActionContext<PayoutsState, RootState>, nodeId: string): Promise<void> {
        const selectedSatelliteId = ctx.rootState.nodes.selectedSatellite ? ctx.rootState.nodes.selectedSatellite.id : null;
        const paystub = await this.payouts.paystub(selectedSatelliteId, ctx.state.selectedPayoutPeriod, nodeId);

        ctx.commit('setNodePaystub', paystub);
    }

    /**
     * nodeTotals action loads total payouts information for selected node.
     * @param ctx - context of the Vuex action.
     * @param nodeId
     */
    public async nodeTotals(ctx: ActionContext<PayoutsState, RootState>, nodeId: string): Promise<void> {
        const paystub = await this.payouts.paystub(null, null, nodeId);

        ctx.commit('setNodeTotals', paystub);
    }

    /**
     * heldHistory action loads held history for selected node.
     * @param ctx - context of the Vuex action.
     * @param nodeId
     */
    public async heldHistory(ctx: ActionContext<PayoutsState, RootState>, nodeId: string): Promise<void> {
        const heldHistory = await this.payouts.heldHistory(nodeId);

        ctx.commit('setNodeHeldHistory', heldHistory);
    }

    // Getters
    /**
     * periodString is full name month and year representation of selected payout period.
     */
    public periodString(state: PayoutsState): string {
        if (!state.selectedPayoutPeriod) { return 'All time'; }

        const splittedPeriod = state.selectedPayoutPeriod.split('-');
        const monthIndex = parseInt(splittedPeriod[1]) - 1;
        const year = splittedPeriod[0];

        return `${monthNames[monthIndex]}, ${year}`;
    }
}
