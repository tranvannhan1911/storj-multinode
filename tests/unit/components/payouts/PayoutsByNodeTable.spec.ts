// Copyright (C) 2021 Storj Labs, Inc.
// See LICENSE for copying information.

import Vuex from 'vuex';
import { createLocalVue, shallowMount } from '@vue/test-utils';

import { Currency } from '@/app/utils/currency';
import { Paystub } from '@/payouts';
import { Size } from '@/private/memory/size';

import PayoutsByNodeTable from '@/app/components/payouts/tables/payoutsByNode/PayoutsByNodeTable.vue';

const localVue = createLocalVue();

localVue.use(Vuex);

localVue.filter('centsToDollars', (cents: number): string => Currency.dollarsFromCents(cents));

localVue.filter('bytesToBase10String', (amountInBytes: number): string => Size.toBase10String(amountInBytes));

describe('PayoutsByNodeTable', (): void => {
    it('renders correctly', (): void => {
        const paystub = new Paystub(3000, 4000, 5000, 6000,
            50000, 60000, 70000, 80000, 90000, 100000, 110000);

        const wrapper = shallowMount(PayoutsByNodeTable, {
            localVue,
            propsData: { paystub },
        });

        expect(wrapper).toMatchSnapshot();
    });
});
