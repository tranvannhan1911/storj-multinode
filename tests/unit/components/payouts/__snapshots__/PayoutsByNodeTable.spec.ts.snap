// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PayoutsByNodeTable renders correctly 1`] = `
<base-table-stub>
  <thead>
    <tr>
      <th class="align-left">NAME</th>
      <th>DISK</th>
      <th>BANDWIDTH</th>
      <th>PAYOUT</th>
    </tr>
  </thead>
  <tbody>
    <tr class="table-item">
      <th class="align-left">Download</th>
      <th>--</th>
      <th>4KB</th>
      <th>$0.06</th>
    </tr>
    <tr class="table-item">
      <th class="align-left">Download Repair</th>
      <th>--</th>
      <th>11KB</th>
      <th>$0.15</th>
    </tr>
    <tr class="table-item">
      <th class="align-left">Disk Space</th>
      <th>3KBm</th>
      <th>--</th>
      <th>$0.05</th>
    </tr>
    <tr class="table-item">
      <th class="align-left">Gross Total</th>
      <th></th>
      <th></th>
      <th>$0.00</th>
    </tr>
    <tr class="table-item">
      <th class="align-left">Held amount</th>
      <th></th>
      <th></th>
      <th>$0.09</th>
    </tr>
    <tr class="table-item">
      <th class="align-left">NET TOTAL</th>
      <th></th>
      <th></th>
      <th>$0.10</th>
    </tr>
    <tr class="table-item">
      <th class="align-left">Distributed</th>
      <th></th>
      <th></th>
      <th>$0.11</th>
    </tr>
  </tbody>
</base-table-stub>
`;
